### DISCUSSION TITLE
как его запустить вообще?

### DISCUSSION THREAD

**Discussion by vovamidas** [bol-van/zapret#902] [Q&A]
я только скачал и у меня папка эта есть. куда че нажимать?
![image](https://github.com/user-attachments/assets/7f7e5f39-1a25-4b56-a9f5-b78cbe4203eb)

**Comment by numbereleven-a** [comment]
https://github.com/bol-van/zapret?tab=readme-ov-file#%D0%B1%D1%8B%D1%81%D1%82%D1%80%D1%8B%D0%B9-%D1%81%D1%82%D0%B0%D1%80%D1%82

**Reply by BlackOwl2110** [reply]
вот смотрю и все равно без понятия что куда запускать что как делать. Так себе руководство по использованию

**Reply by icynstant** [reply]
@BlackOwl2110 https://github.com/bol-van/zapret/blob/master/docs/quick_start_windows.md

**Comment by electrifying** [comment]
Не там скачали, однокнопочные батники в другом репозитории

**Comment by Rand0mizer777** [comment]
> я только скачал и у меня папка эта есть. куда че нажимать? ![image](https://private-user-images.githubusercontent.com/190259054/391322105-7f7e5f39-1a25-4b56-a9f5-b78cbe4203eb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc3NzU1OTAsIm5iZiI6MTc1Nzc3NTI5MCwicGF0aCI6Ii8xOTAyNTkwNTQvMzkxMzIyMTA1LTdmN2U1ZjM5LTFhMjUtNGI1Ni1hOWY1LWI3OGNiZTQyMDNlYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxM1QxNDU0NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMWFiYmQ3Mjg2MjFmNGFiNDNlNjk0MDZkOTk5OGNhNTZjMzg5MDg2ZjU4M2M3NThhYmI0M2ExOTY5NjUyODJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MtZsAa6d9o0ZZyTImKSR2gB5UAgdUDPUQ0hYRadGHMQ)

Если на винду требуется, то здесь нажимаете главную страницу zapret и ищете быстрый старт виндоус, если на роутер, то у меня для вас плохие новости.

**Comment by Avatar-Lion** [comment]
Вам сюда: https://github.com/bol-van/zapret/blob/master/docs/quick_start_windows.md

**Comment by andreydegt** [comment]
Непонятно совершенно как его запускать. Внимательно читал квикстарт и вообще. Установил на linux mint.  install_bin.sh, install_prereq.sh.
blockcheck.sh. В мануале нет вообще ничего, куда идет выход этого blockcheck.sh. Учиться телепатии? Ковыряться в тексте скриптов?
После install  что происходит? Где бинарники? Устанавливается ли служба? Если да, как она зовется? Почему это не написать в мануалах, как в других проектах?

**Reply by HentayHentay** [reply]
1 - проект нацелен на более-менее опытного пользователя
2 - иногда придется писать/создавать  руками
3 - мануал написан как он есть. Не можете понять, вам к сборщикам или к VPN/VDS 
4 - не зря помимо квика, присутсвует и полный мануал. Не понятно, читаем полный. Быстрый для опытных 

> Установил на linux mint. install_bin.sh, install_prereq.sh.

Не установили, лишь частично завели запрет для тестирования его работы. Для установки  *_easy.sh

> . В мануале нет вообще ничего, куда идет выход этого blockcheck.sh

Вывод блокчека используется для написания стратегии, что заносится в конфиг запрета.


> После install что происходит? Где бинарники?

По умолчанию в папке /opt/zapret/

> Устанавливается ли служба? Если да, как она зовется?

она зовется zapret . В зависимости от демона, отвечающие за обработку служб (init.d/systemd) помещается в папку данного демона 
В minte используется systemd, а значит в консоле можно вызвать список служб и там посмотреть.

> Почему это не написать в мануалах, как в других проектах?

Возвращаемся к п.1 и п.4

**Reply by andreydegt** [reply]
> Не можете понять
Единственно, чего я не могу понять, почему внятно не написать, что делает каждый скрипт. Чтобы тому, кто пользуется проектом, не пришлось несколько часов работать следопытом.
До этого мне вполне хватало понимания устанавливать десятки различных утилит из исходников в том числе. Писать свои. Писать юниты служб. Потому что авторы проектов писали человеrообразные md.
Вот здесь быстрый старт
https://github.com/bol-van/zapret/blob/master/docs/quick_start.md
Где там хоть слово, куда выводится результат blockcheck.sh. , в какой файл?
Оттуда
> Откроется редактор, куда впишите созданную на предыдущем этапе стратегию.
Как я должен догадаться, где мне взять эту "созданную на предыдущем этапе стратегию."

**Reply by HentayHentay** [reply]
Все написано в полном мануале. Квик - для опытных 

> Где там хоть слово, куда выводится результат blockcheck.sh. , в какой файл?

В линуксе помимо записи в файл есть стандартный вывод. Он не пишеться в файл. 
У блокчека есть консольные переменные, которые есть в полном мануале. Его можно перенаправить после настройки и передачи переменных для записи в файл


> Откроется редактор, куда впишите созданную на предыдущем этапе стратегию.

В линуксе много редакторов. Вопрос в том, чем будете редактировать конфиг запрета, для записи стратегии. 



> Как я должен догадаться, где мне взять эту "созданную на предыдущем этапе стратегию."

В полном мануале описано все это. Но мы специально идем коротким путем.

Сокращенно:
Блокчек -> получили стратегию -> редактируем конфиг запрета -> сохраняемся -> перезапускаем запрет -> радуемся полученному результату

**Reply by andreydegt** [reply]
> В линуксе помимо записи в файл есть стандартный вывод. Он не пишеться в файл.
> У блокчека есть консольные переменные, которые есть в полном мануале. Его можно перенаправить после настройки и передачи переменных для записи в файл

Вот и я об этом. В линуксе много чего есть. И читая квикстарт bolvan zapret впору бросать гадальные кости. Хотя можно было написать ясно и понятно. При владении родным языком. См. выше.

**Comment by andreydegt** [comment]
Кстати нашел мануал, где нет лишних слов, только полезные. И в правильной последовательности. Что где, куда и зачем. Все понятно, хоть и на английском. И все заработало.
https://fr0stb1rd.gitlab.io/posts/installing-zapret-on-linux-a-practical-guide-to-bypassing-censorship/

**Reply by HentayHentay** [reply]
Браво.. Поиск работает!!!

PS Хотя в полном описание все действия присутствует. Вы сами с усами)
